bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Device.java

```diff
@@ -55,6 +58,7 @@ public abstract class Device implements Drawable {
 	String[] loadedFonts;
 
 	volatile boolean disposed;
+	private Vector<Resource> resourcesWithZoomSupport  = new Vector<>();
```

Comment:
Why use outdated Vector instead of, e.g., a synchronized set?

Reply:
Done

Reply:
I meant to use a synchronized set, not to have a set to which every access it synchronized (as that's quite similiar to using a vector).

Reply:
Oh sorry, its done now

Reply:
I suppose everything has to happen in the UI thread so do we need synchronized (whatever) anyways?
That's actually one of the main points why UI frameworks use "display threads" to prevent the synchronization and possible deadlocks if multiple threads would operate on the same output device.

Reply:
Valid point. I did not challenge the original answer that synchronization is necessary here (see https://github.com/eclipse-platform/eclipse.platform.swt/pull/1477#discussion_r1783956150), but taking a look at the implementation I do not see any access to the data structure that can happen from outside the UI thread, as both resource instantiation and OS callback processing happen within the UI thread (i.e., the thread of the according `Device`/`Display` instance), don't they?